osu.Game.Rulesets.Osu/Mods/OsuModApproachDifferent.cs

```diff
@@ -11,14 +12,16 @@
 
 namespace osu.Game.Rulesets.Osu.Mods
 {
-    public class OsuModApproachDifferent : Mod, IApplicableToDrawableHitObject
+    public class OsuModApproachDifferent : Mod, IApplicableToDrawableHitObject, IMutateApproachCircles
     {
         public override string Name => "Approach Different";
         public override string Acronym => "AD";
         public override string Description => "Never trust the approach circles...";
         public override double ScoreMultiplier => 1;
         public override IconUsage? Icon { get; } = FontAwesome.Regular.Circle;
 
+        public override Type[] IncompatibleMods => new[] { typeof(IMutateApproachCircles) };
```

Comment:
A little unfortunate that implementing the interface is not enough and this spec has to exist, but alas...

Maybe this can be enforced with a test at least? Something like, find all implementors of `IMutateApproachCircles` in this project, and ensure they're all pairwise-incompatible?

Reply:
To make sure there is symmetry between interface application?

Sounds good, but probably part of a larger test suite to cover all mods, would rather get this change in today with minimal changes (before a release tonight or tomorrow).

Reply:
> To make sure there is symmetry between interface application?

Yeah that would be the intention. The alternative would be to somehow mark the interface itself as mutually exclusive and handle that in the incompatibility logic but I'm not sure we want more complexity there, which is why I didn't suggest it.

> Sounds good, but probably part of a larger test suite to cover all mods, would rather get this change in today with minimal changes (before a release tonight or tomorrow).

Ok, fair.